### PR TITLE
doc: add tutorial for multi-signer script execution

### PR DESCRIPTION
**Pull request by neutrinoks**
- added a new tutorial as an extension to the existing one for the case of multi-signer execution requests
- updated the final-design document regarding changed RPC interfaces
- added a new conceptional chapter about multi-signer execution requests to the final-design document
- updated pallet integration steps in tech-guide document regarding the multi-signer feature
- updated and added helper scripts for the multi-signer move-project example

### CODE REVIEW

File: doc/tutorial-multi-signer.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+After all three students have executed that script transaction you should see, that the amounts of tokens have be withdrawn from their accounts.
+
+Because every student has to sign the extrinsic call and execute the script execution, the execution request will be stored in the pallet's storage until every student has signed.
+> Note: After the initiation of a multi signer execution requests, the specified cheque limits will be locked until it the request gets executed or deleted.
+From the first signer to the one before the last one, they will see the event `SignedMultisigScript`.
+After the final signer has signed and executed the same extrinsic call, the script transaction will be executed, tokens with amount of the cheque limit will be unlocked and withdrawn, and the event `ExecuteCalled` will be emitted.
+Here, the second was Dave, and the last one was Eve, who signed the extrinsic call. As a result you can see the events in that order.
+| ![polkadot.js_multisign_events.png](assets/polkadot.js_multisign_events.png) |
+|:--:|
+| _Events in a multi signer execution requests in [polkadot.js][polkadotjs]_ |
```
Comment (Rqnsom) [inline doc/tutorial-multi-signer.md:36]: In the end, we can add an outro message and say something about how if they want to know more, they can check our `Multi Signer Script Execution` chapter in the final design.md doc.

**Review by Rqnsom — APPROVED** [review]
LGTM 🚀

File: doc/final-design.md [hunk 2]
```diff
@@ -186,3 +183,17 @@ How it works under the hood is shown in a simple UML diagram below:
...
+When the first signer executes a script transaction with multiple signers, the Move pallet will create a multi-signer execution request in the storage.
+Each following signer will execute the same script transaction with its individual cheque limit for the planned script execution.
+Pallet move will store the state of each signature and each cheque limit and keep track if all users have signed or not.
+After each signer has signed by calling the exact same extrinsic call with the same script transaction, the Move pallet will execute the script.
+
+**Differences to single signer:**
+- The cheque limit (tokens) of each signer will be locked on their accounts until the request gets finally executed or deleted.
+- Except for the final signer, the event `SignedMultisigScript` will be emitted instead of `ExecuteCalled`.
+- When all signers have signed, the script will be executed, the tokens be unlocked, and balances applied according to the Move script.
+- In case not every signer signes the execution request, the request gets removed automatically after a certain amount of time (defined by the blockchain developer).
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply doc/final-design.md:199]: Let's modify the first sentence of the last bullet point a bit. To me, it sounds like the first signer could choose/define the timeout, but he can only choose the point of time when he signs... (that will be hard for some people to understand).

File: doc/final-design.md [hunk 3]
```diff
@@ -186,3 +183,17 @@ How it works under the hood is shown in a simple UML diagram below:
...
+
+# Multi Signer Script Execution
+
+Executing Move scripts with multiple signers works basically the same as for a single signer.
+When the first signer executes a script transaction with multiple signers, the Move pallet will create a multi-signer execution request in the storage.
+Each following signer will execute the same script transaction with its individual cheque limit for the planned script execution.
+Pallet move will store the state of each signature and each cheque limit and keep track if all users have signed or not.
+After each signer has signed by calling the exact same extrinsic call with the same script transaction, the Move pallet will execute the script.
+
+**Differences to single signer:**
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply doc/final-design.md:195]: Done

File: doc/final-design.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (neutrinoks) [reply doc/final-design.md:199]: Updated, with slight modifications

File: doc/tech_guide.md [hunk 4]
```diff
@@ -75,11 +75,27 @@ Read detailed info about test concepts (what tests cover and how) in the [old te
 
 ## Pallet Configuration in a Substrate-Node
 
-The pallet's configuration is concise. Besides the regular `RuntimeEvent` and a predefined `WeightInfo`, you only have to tell the pallet about your `Currency` handler:
+The pallet's configuration is concise. Besides the regular `RuntimeEvent` and a predefined `WeightInfo`, you have to tell the pallet about your `Currency` handler, the used balance data type in your blockchain (`CurrencyBalance`), the maximum lifetime (`MaxLifetimeRequests`) and maximum number of signers (`MaxScriptSigners`) in case of a multi signer script execution request:
 ```rust
+parameter_types! {
+    // Number of blocks after that a multi signer request gets removed.
+    pub const MaxLifetimeRequests: BlockNumberFor<Test> = 5;
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply doc/tech_guide.md:82]: Like it :+1: Yeah, let's change this to this name after this PR :+1: I will add a todo here to remember this good suggestion

File: doc/tutorial-multi-signer.md (hunk 1 above)
Comment (neutrinoks) [reply doc/tutorial-multi-signer.md:37]: Done

File: doc/final-design.md [hunk 5]
```diff
@@ -186,3 +183,17 @@ How it works under the hood is shown in a simple UML diagram below:
...
+- The cheque limit (tokens) of each signer will be locked on their accounts until the request gets finally executed or deleted.
+- Except for the final signer, the event `SignedMultisigScript` will be emitted instead of `ExecuteCalled`.
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply doc/final-design.md:197]: Done

**Pull request merged**